os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -282,58 +308,114 @@ static int binary_manager_load_all(void)
 	return BINMGR_OPERATION_FAIL;
 }
 
-static void reload_kill_each(FAR struct tcb_s *tcb, FAR void *arg)
+static int binary_manager_kill_task(FAR struct tcb_s *tcb)
+{
+	irqstate_t saved_state;
+
+#if defined(CONFIG_SCHED_ATEXIT) || defined(CONFIG_SCHED_ONEXIT)
```

Comment:
We cannot use `task_atexit` without CONFIG_SCHED_ATEXIT. It is also same as `task_onexit`. Please check the condition.